taichi/backends/opengl/opengl_api.cpp

```diff
@@ -390,6 +391,8 @@ struct CompiledProgram::Impl {
     }
     for (const auto &ker : kernels) {
       auto guard = launcher->create_launch_guard(iov);
+      if (ker->rse.has_value()) ker->num_groups = (*ker->rse)((const char *)gtmp_base);
+      TI_DEBUG("kernel [{}] num_groups = {}", ker->kernel_name, ker->num_groups);
```

Comment:
I specify **dynamic work group size** for dyn-range-kernels (thanks to the GL feature enable me).
Is this better than grid-stride-loop?
@k-ye @yuanming-hu Does other backends enables us to dynamically change work group size? If so we may make use of that?

Reply:
For metal, I don't think so .

Reply:
For CUDA, I guess the answer is no. A seemingly related feature is https://devblogs.nvidia.com/cuda-dynamic-parallelism-api-principles/ 

My feeling is grid-strided loops are usually good enough.